### ISSUE TITLE
Looking for maintainers

### ISSUE THREAD

**Issue by mattyclarkson** [cpplint/cpplint#49]
I personally have moved into working with TypeScript and Rust so no longer daily work on C++. I still have an interest in the project as I feel that `cpplint` provides a great service to the C++ toolchain that is easily available in other languages.

If there are other people that use `cpplint` daily and are interested in moving the project forward get in touch on this issue.

**Comment by KYSXD** [comment]
How can one help?

**Comment by mattyclarkson** [comment]
Reviewing PRs from @tkruse would really help right now. That's one of the best ways to help.

**Comment by tkruse** [comment]
@KYSXD: If you can review more open PRs, that may help

**Comment by tkruse** [comment]
@mattyclarkson , I cannot seem to reach you by mail. If you could add me to the organization, I can go ahead with my PRs.

**Comment by mattyclarkson** [comment]
@tkruse made you owner. Have at it.

**Comment by tkruse** [comment]
@jayvdb, @cclaus, any interest in helping maintaining this fork? I don't expect much time investment, mostly I believe more people should have access rights to the repo and pypi accounts in case of emergencies.

**Comment by aaronliu0130** [comment]
@mattyclarkson @tkruse While I don't code much Python, I understand the basic syntax of it and I'm willing to process PRs and tags. I also use cpplint daily.

If somebody's checking, yes, you can still ask to become maintainer! A third person is always welcome.

**Comment by jayvdb** [comment]
hi @tkruse & @mattyclarkson , sorry for not seeing this earlier.  I am also mostly using Rust these days, but I still use cpplint as part of the CI of https://github.com/py2many/py2many which I actively maintain.  I am happy to help out with cpplint to get the backlog down.

**Comment by tkruse** [comment]
@aaronliu0130 and @jayvdb , let me give you repo access, I cannot seem to find time to even respond to issues right now.

**Comment by aaronliu0130** [comment]
Thanks! I think I'll ask for PyPI only when I need it :)

**Comment by aaronliu0130** [comment]
@tkruse Also, what's the meaning of developing on develop instead of master? We already have a separate release branch and a PR just merged into master. After finishing the pull requests, can I merge master into develop and delete master?

**Comment by jayvdb** [comment]
At the moment, please only merge PRs onto `develop`, and don't touch the other branches until we've learned and documented the purposes of the other existing branches.

I've seen https://github.com/cpplint/cpplint/commits/master/ and https://github.com/cpplint/cpplint/commits/develop/ and the recent merges are failing CI.  We need to ensure CI is working before merging anything else.

**Comment by aaronliu0130** [comment]
Yeah, I'm trying to do so. The only failure seems to be because of something with 3.7. However, setup.py is also outdated, so we'd either downgrade setuptools or convert it into some TOML. I'm working on the CI in [my fork](https://github.com/aaronliu0130/cpplint/actions/runs/7680223101/job/20931999774).

**Comment by aaronliu0130** [comment]
I've managed to get this to work except for 3.11+. Let's do that for now

**Comment by aaronliu0130** [comment]
@tkruse Sorry for what's probably the current space of your inbox :p, but I'd just like to know what formatter and formatting style we used.

**Comment by tkruse** [comment]
Wow, you guys are so quick. So I have been using `develop` branch to try out on CI before merges  mostly, trying to merge to master only when develop branch is stable. Things have a way of working fine on local but breaking on CI every now and then (usually because CI infrastructure changed), and I like my master branch going from green to green in CI, that's all.


The CONTRIBUTING file should explain a bit. This project is mostly dead and written in stone, as we try to be prepared for any future additional public releases of Googles internal version (any minor change we made to this version here will be annoying if such a merge ever comes, even if that might never be). So maintenance here is mostly very tiny fixes that should be minimal in terms of changed lines.

For any code that we write, I am only trying to keep the static checkers happy, else I would suggest to try to stay consistent with all the code that we do not want to touch, even if the code style is not what we prefer.

**Comment by aaronliu0130** [comment]
Thanks! Just to confirm, by "static checkers", do you mean pylint, flake8 and the like?

And to summarize, merge develop's commits periodically if CI succeeds? And, I assume all commits with nothing ahead of master can be deleted?

Also, would it be worth it to convert the code to Google's style guide someday?

**Comment by tkruse** [comment]
Yes, static checkers as those.

I would recommend to use the develop branch only just before making a new release, though I might not always have done so. Commits in git do not get deleted. Either the develop branch is rebased to master after merging, or master is just tracking develop (without merge commits).
So outside of release activity, develop should look same as master, then some PRs and commits are added to develop, and when CI says it looks good, the changes move to master, via merge or fast-forward. It's only the result that counts for me, this is no high-activity project that needs more guardrails I think.

As I said, I recommend not making any broad changes like Styleguide changes to cpplint, there are still people making PRs against the upstream repo, so staying close to that repo has more benefits than Python Styleguide adherence.

**Comment by aaronliu0130** [comment]
By "all commits" I meant branches, lol

**Comment by jayvdb** [comment]
https://github.com/cpplint/cpplint/commits/develop/ and https://github.com/cpplint/cpplint/commits/master/ are an absolute mess.

@tkruse  , please update the settings to force all changes to have a PR , that passes CI, and has been reviewed by someone other than the author.

And I guess one of those branches needs to be locked so we dont have two concurrent branches being developed on.

**Comment by aaronliu0130** [comment]
I agree that we should probably limit ourselves to one of the branches and force PRs to pass CI. However, outside of the merge commits and related duplicated commits, I don't see how the branch is a mess. I think smaller changes that don't impact much should be pushed to master as long as tests are passed locally. Requiring all changes to be made via PRs adds bureaucracy and clutters up the view of closed PRs.

**Comment by aaronliu0130** [comment]
I'd recommend deleting master and renaming develop to master.

**Comment by tkruse** [comment]
@jayvdb , I have added some rules to master and develop now, I hope it helps

**Comment by tkruse** [comment]
@jayvdb , please do feel free to improve the states of the main branches. if necessary, hard reset the master branch to a state that you feel comfortable about, and restart merging from there.

**Comment by aaronliu0130** [comment]
I'm pretty sure we can't rebase master as it's a protected branch.

**Comment by tkruse** [comment]
@jayvdb should now have permission to rewrite master branch, and I see no much issues doing so as long as already released commits remain untouched.

**Comment by aaronliu0130** [comment]
Hi @jayvdb, as I've said in email #268 really needs to be merged.

**Comment by aaronliu0130** [comment]
Hey @jayvdb, is it possible for you to add someone else you know at least?

**Comment by aaronliu0130** [comment]
Hey, @jayvdb, I've taken to using an alt account to do critical fixes and documentation updates.

**Comment by aaronliu0130** [comment]
Status: We're currently in a weird situation where we only have collaborator perms, as the repo belongs to the cpplint organization, which consists of @mattyclarkson (and used to also include Andrew Davis) and practically also includes tkruse. This means that we can't add app integrations, add new maintainers, or change branch protection rules—which now include the review requirement after my own idiocy sparked jayvdb to ask tkruse to change it—unless we ask mattyclarkson or tkruse (who also has the permissions). I've sent an email to mattyclarkson and theandrewdavis requesting membership already, so hopefully a ping here would help.

**Comment by mattyclarkson** [comment]
What do you need? I'm happy to switch up membership then sunset myself as I don't have the time for involvement anymore.

**Comment by aaronliu0130** [comment]
It would be great if you could add me, jayvdb, and @cclauss to the cpplint organization.

**Comment by cclauss** [comment]
Yes, I am willing to join.  Unfortunately my username was misspelled above or I would have been more involved years ago.
* https://github.com/cpplint/cpplint/issues/49#issuecomment-466743685 one `s` instead of two in @cclauss.

**Comment by aaronliu0130** [comment]
@mattyclarkson What do you think?
@jayvdb Could you also give cclauss or me access to our PyPI?

**Comment by aaronliu0130** [comment]
I've found a config change we could use: requiring CI to pass.

**Comment by aaronliu0130** [comment]
Hey @tkruse, it's taking some time to release the next version of cpplint. Could you give cclauss or me the maintainer perms for PyPI? My PyPI username is aliu, and I'm fairly sure cclauss's is cclauss.  Thanks in advance!
@mattyclarkson Are you still available to add us to the organization? Thanks in advance!

**Comment by jayvdb** [comment]
I'll be providing @cclauss PyPI maintainer soon.

**Comment by cclauss** [comment]
I would like to follow the instructions at https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows to set up a GitHub Actions workflow for publishing new releases to PyPI.  https://pypi.org/project/cpplint/
 
This release workflow is reliable for other PyPI projects that I co-maintain.

**Comment by cclauss** [comment]
After reviewing PyPI settings with @jayvdb yesterday, he does not have PyPI Manage permissions so he cannot set up Trusted Publishing and cannot add new project maintainers.

Are any of the original maintainers willing to adjust the PyPI configuration?

**Comment by aaronliu0130** [comment]
Oh dear, that's not good.

@mattyclarkson @tkruse

**Comment by aaronliu0130** [comment]
Unfortunately I think we need to make the next release without trusted publishing and perhaps we need jayvdb to release it after all. Hopefully @mattyclarkson @tkruse or @theandrewdavis would respond, but otherwise, we could make a new organization and PyPI listing if needed.

**Comment by mattyclarkson** [comment]
Added as owners. I will remove myself once at least one of you has accepted.

cc: @tkruse

**Comment by cclauss** [comment]
Great.  Thanks but on PyPI (https://pypi.org/project/cpplint) @jayvdb is a ___Maintainer___ but not a ___Manager___.

**Comment by jayvdb** [comment]
I dont see an invite to be a manager.

**Comment by mattyclarkson** [comment]
Now owner.

**Comment by aaronliu0130** [comment]
Thanks mattyclarkson!

**Comment by jayvdb** [comment]
Hi @mattyclarkson , could you also grant me manager on https://test.pypi.org/project/cpplint/ ?

This is relevant4 to https://github.com/cpplint/cpplint/pull/347 again

**Comment by mattyclarkson** [comment]
I though you already were, there is already a user I have set to owner with your name that was previously set as maintainer. What is your PyPI username?

**Comment by cclauss** [comment]
https://test.pypi.org/user/John.Mark.Vandenberg -- This is what we need for #347 to pass.
https://pypi.org/user/John.Mark.Vandenberg

**Comment by aaronliu0130** [comment]
To clarify, we have the perms on pypi but not test.pypi.

**Comment by cclauss** [comment]
To clearify, @jayvdb already has sufficient perms on PyPI but #347 is currently failing on testPyPI because he does not have sufficient prems there.

**Comment by aaronliu0130** [comment]
Let's migrate to organization discussions: It's quite unwieldy to put all of our meta discussion in one issue.
